http-auth/routes/routes.go

```diff
@@ -0,0 +1,109 @@
+package routes
+
+import (
+	"fmt"
+	"html"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/joho/godotenv"
+	"golang.org/x/time/rate"
+)
+
+const HTMLHeader = "<!DOCTYPE html><html>"
+
+func IsTestRun() bool {
+	return os.Getenv("EXECUTION_ENVIRONMENT") == "test"
+}
+
+func loadEnv() {
+	path := ""
+	if IsTestRun() {
+		path = "../.env"
+	}
+	err := godotenv.Load(path)
+	if err != nil {
+		log.Fatal("Error loading .env file")
+	}
+}
+
+func IndexHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "GET" {
+		w.Header().Set("Content-Type", "text/html")
+		b := "<em>Hello World</em>"
+
+		queryParams := r.URL.Query()
+
+		if len(queryParams) > 0 {
+			b = "<h1>Query Params</h1><ul>"
+			for key, value := range queryParams {
+				b += fmt.Sprintf("<li>%s: %s</li>", key, html.EscapeString(strings.Join(value, ", ")))
```

Comment:
How do you know that `key` doesn't need escaping too?

Reply:
I don't :)